delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/ResourcesDummy.scala

```diff
@@ -23,88 +24,112 @@ import monix.bio.{IO, Task, UIO}
-    sourceAsJsonLD(id, source).flatMap { case (compacted, expanded) =>
-      eval(CreateResource(id, project, schema, source, compacted, expanded, caller))
-    }
+    for {
+      jsonld               <- sourceAsJsonLD(id, source)
+      (compacted, expanded) = jsonld
+      project              <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      res                  <- eval(CreateResource(id, projectRef, schema, source, compacted, expanded, caller), project.apiMappings)
+    } yield res
 
   override def update(
       id: Iri,
-      project: ProjectRef,
+      projectRef: ProjectRef,
       schemaOpt: Option[ResourceRef],
       rev: Long,
       source: Json
   )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
-    sourceAsJsonLD(id, source).flatMap { case (compacted, expanded) =>
-      eval(UpdateResource(id, project, schemaOpt, source, compacted, expanded, rev, caller))
-    }
+    for {
+      jsonld               <- sourceAsJsonLD(id, source)
+      (compacted, expanded) = jsonld
+      project              <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      mapping               = project.apiMappings
+      res                  <- eval(UpdateResource(id, projectRef, schemaOpt, source, compacted, expanded, rev, caller), mapping)
+    } yield res
 
   override def tag(
       id: Iri,
-      project: ProjectRef,
+      projectRef: ProjectRef,
       schemaOpt: Option[ResourceRef],
       tag: Label,
       tagRev: Long,
       rev: Long
   )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
-    eval(TagResource(id, project, schemaOpt, tagRev, tag, rev, caller))
+    for {
+      project <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      res     <- eval(TagResource(id, projectRef, schemaOpt, tagRev, tag, rev, caller), project.apiMappings)
+    } yield res
 
   override def deprecate(
       id: Iri,
-      project: ProjectRef,
+      projectRef: ProjectRef,
       schemaOpt: Option[ResourceRef],
       rev: Long
   )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
-    eval(DeprecateResource(id, project, schemaOpt, rev, caller))
+    for {
+      project <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
+      res     <- eval(DeprecateResource(id, projectRef, schemaOpt, rev, caller), project.apiMappings)
+    } yield res
 
-  override def fetch(id: Iri, project: ProjectRef, schemaOpt: Option[ResourceRef]): UIO[Option[DataResource]] =
-    journal
-      .currentState((project, id), Initial, Resources.next)
-      .map(_.flatMap(_.toResource))
-      .map(validateSameSchema(_, schemaOpt))
+  override def fetch(
+      id: Iri,
+      projectRef: ProjectRef,
+      schemaOpt: Option[ResourceRef]
+  ): IO[ResourceRejection, Option[DataResource]] =
+    for {
+      project  <- projects.activeProject(projectRef).leftMap(WrappedProjectRejection)
```

Comment:
We should be able to fetch the resource of a deprecated project.
Here also, we may want to have a specific revision

Reply:
Also we start with fetching the resource, it means the fetch of the project should not fail

Reply:
No specific revision of project, but yes you're right, you can still fetch resources on deprecated projects